### PR TITLE
chore: Hedera app compiler warning fixes

### LINKED ISSUE #15561: Fix compiler warnings in hedera-app

### PR DESCRIPTION
**Pull request by derektriley**
**Description**:

Fixes compilation warnings in hedera-app

**Related issue(s)**:

Fixes #15561 

**Notes for reviewer**:

**Checklist**

- [ ] Documented (Code comments, README, etc.)
- [ ] Tested (unit, integration, etc.)

### CODE REVIEW

**Review by petreze — DISMISSED** [review]

File: hedera-node/hedera-app/build.gradle.kts [hunk 1]
```diff
@@ -23,12 +23,6 @@ plugins {
 
 description = "Hedera Application - Implementation"
 
-// Remove the following line to enable all 'javac' lint checks that we have turned on by default
-// and then fix the reported issues.
-tasks.withType<JavaCompile>().configureEach {
-    options.compilerArgs.add("-Xlint:-exports,-lossy-conversions,-cast")
```
Comment (san-est) [inline hedera-node/hedera-app/build.gradle.kts:29]: Should we comment them out only if there is a need to disable again?

**Review by mhess-swl — APPROVED** [review]

**Review by mhess-swl — APPROVED** [review]

**Review by thomas-swirlds-labs — DISMISSED** [review]
LGTM @derektriley 🎉

**Review by petreze — DISMISSED** [review]

**Review by mhess-swl — DISMISSED** [review]

**Review by rbarkerSL — DISMISSED** [review]

**Review by kimbor — DISMISSED** [review]

**Review by mhess-swl — APPROVED** [review]

**Review by tinker-michaelj — APPROVED** [review]
LGTM, ty @derektriley !

**Review by rbarkerSL — APPROVED** [review]
Approval applies to

- `hedera-node/hedera-app/build.gradle.kts`
- `hedera-node/hedera-app/src/main/java/module-info.java`
- `hedera-node/hedera-token-service-impl/src/main/java/module-info.java`
- `hedera-node/hedera-token-service/src/main/java/module-info.java`

**Pull request merged**